bundle/ScheduledVisibility/Handler/ObjectState.php

```diff
@@ -0,0 +1,133 @@
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Handler;
+
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\ObjectStateService;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\ObjectState\ObjectStateGroup;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+use function sprintf;
+
+final class ObjectState implements ScheduledVisibilityInterface
+{
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ObjectStateService $objectStateService,
+        private readonly int $objectStateGroupId,
+        private readonly int $hiddenObjectStateId,
+        private readonly int $visibleObjectStateId,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {}
+
+    public function hide(Content $content): void
+    {
+        try {
+            if ($this->isHidden($content)) {
+                return;
+            }
+        } catch (NotFoundException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Configured object state group does not exist: %s',
+                    $e->getMessage(),
+                ),
+            );
+        }
+
+        $hiddenObjectStateId = $this->hiddenObjectStateId;
+
+        $this->toggleObjectState($content, $hiddenObjectStateId);
+    }
+
+    public function reveal(Content $content): void
+    {
+        try {
+            if (!$this->isHidden($content)) {
+                return;
+            }
+        } catch (NotFoundException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Configured object state group does not exist: %s',
+                    $e->getMessage(),
+                ),
+            );
+        }
+
+        $visibleObjectStateId = $this->visibleObjectStateId;
+
+        $this->toggleObjectState($content, $visibleObjectStateId);
+    }
+
+    public function getType(): HandlerType
+    {
+        return HandlerType::ObjectState;
+    }
+
+    /**
+     * @throws NotFoundException
+     */
+    public function isHidden(Content $content): bool
+    {
+        $objectStateGroupId = $this->objectStateGroupId;
+        $objectStateGroup = $this->repository->sudo(
+            fn (): ObjectStateGroup => $this->objectStateService->loadObjectStateGroup($objectStateGroupId),
+        );
+
+        $objectState = $this->repository->sudo(
+            fn (): \Ibexa\Contracts\Core\Repository\Values\ObjectState\ObjectState => $this->objectStateService->getContentState($content->contentInfo, $objectStateGroup),
+        );
+
+        $hiddenObjectStateId = $this->hiddenObjectStateId;
```

Comment:
You can get rid of the temporary variable and inline it into the `return` statement.

Reply:
fixed in https://github.com/netgen/ibexa-scheduled-visibility/pull/1/commits/79273b7c94dd8f702991ffc5d1e9d502d3548c6e